src/App.tsx

```diff
@@ -1,24 +1,200 @@
+          ];
+        });
+      })
+      .catch(() => {
+        setErrorMessage(ErrorAction.ADD);
+        setTimeout(() => {
+          setErrorMessage(ErrorAction.NONE);
+        }, 3000);
+      })
+      .finally(() => {
+        setDisabledInput(false);
+        setTempTodo(undefined);
+      });
+  };
+
+  const deleteTodo = (id: number): Promise<void> => {
+    return new Promise((resolve, reject) => {
+      setLoadingTodoId(prev => [...prev, id]);
+      removeTodo(id)
+        .then(() => {
+          setTodos(todos.filter(todo => todo.id !== id));
+          setLoadingTodoId(prev => prev.filter(todoId => todoId !== id));
+          resolve();
+        })
+        .catch(() => {
+          setErrorMessage(ErrorAction.DELETE);
+          setTimeout(() => {
+            setErrorMessage(ErrorAction.NONE);
+          }, 3000);
+          setLoadingTodoId(prev => prev.filter(todoId => todoId !== id));
+          reject();
+        });
+    });
+  };
+
+  const updateTodo = (updatedTodo:Todo) => {
+    setLoadingTodoId([updatedTodo.id]);
+
+    return changeTodo(updatedTodo)
+      .then(() => {
+        setTodos(current => {
+          return current.map(todo => (
+            todo.id === updatedTodo.id ? updatedTodo : todo
+          ));
+        });
+      })
+      .catch(() => {
+        setErrorMessage(ErrorAction.UPDATE);
+        setTimeout(() => {
+          setErrorMessage(ErrorAction.NONE);
+        }, 3000);
+      })
+      .finally(() => {
+        setLoadingTodoId([]);
+      });
+  };
+
+  const toogleAll = () => {
+    const isNoActiveTodo = todos.filter(todo => todo.completed)
+      .length === todos.length;
+
+    todos.forEach(todo => {
+      if (!todo.completed || isNoActiveTodo) {
+        updateTodo({ ...todo, completed: !todo.completed });
+      }
+    });
+  };
+
+  const deleteTodoCompleted = () => {
+    const completed = todos.filter(todo => todo.completed);
+    const uncompleted = todos.filter(todo => !todo.completed);
+
+    completed.forEach(todo => {
+      deleteTodo(todo.id)
+        .then(() => {
+          setTodos(uncompleted);
+        });
+    });
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          onSubmit={addTodo}
+          isDisabled={disabledInput}
+          errorInput={() => {
+            setErrorMessage(ErrorAction.TITLE);
+            setTimeout(() => {
+              setErrorMessage(ErrorAction.NONE);
+            }, 3000);
+          }}
```

Comment:
it is bad practise to make anonymous arrow functions in JSX
better to create function in component and pass reference here